frontend/internal-packages/agent/src/checkpoint/SupabaseCheckpointSaver.ts

```diff
@@ -777,18 +798,30 @@ export class SupabaseCheckpointSaver extends BaseCheckpointSaver<number> {
     checkpointNs: string,
     values: Record<string, unknown>,
     versions: ChannelVersions,
+    checkpointChannelVersions: Record<string, number | string>,
   ): Promise<Database['public']['Tables']['checkpoint_blobs']['Insert'][]> {
-    if (!values || Object.keys(versions).length === 0) {
+    if (!values) {
       return []
     }
 
+    const versionEntries = new Map(Object.entries(versions))
+
+    if (values[TASKS] !== undefined && !versionEntries.has(TASKS)) {
+      const existingVersion = checkpointChannelVersions[TASKS]
+      const resolvedVersion =
+        existingVersion !== undefined
+          ? existingVersion
+          : this.getNextVersion(undefined)
+      versionEntries.set(TASKS, resolvedVersion)
+    }
+
     const blobs: Database['public']['Tables']['checkpoint_blobs']['Insert'][] =
       []
 
-    for (const [channel, version] of Object.entries(versions)) {
+    for (const [channel, version] of versionEntries) {
       const value = values[channel]
 
-      if (value === undefined || value === null) {
```

Comment:
One major cause was that null was excluded here. In the `branch:to:*` channel used by LangGraph's edge, null can sometimes be the value. The problem was that it wasn't being saved.